Sources/Core/AWSClientRuntime/AWSClientConfigDefaultsProvider.swift

```diff
@@ -100,4 +105,12 @@ public class AWSClientConfigDefaultsProvider {
 
         return RetryStrategyOptions(maxRetriesBase: resolvedMaxAttempts - 1, rateLimitingMode: resolvedRateLimitingMode)
     }
+
+    public static func authSchemes(_ clientName: String) -> [ClientRuntime.AuthScheme] {
+        var supportedAuthSchemes: [ClientRuntime.AuthScheme] = [SigV4AuthScheme()]
+        if clientsThatUseSigV4A.contains(clientName) {
```

Comment:
I'm not the biggest fan of this runtime resolution of default auth schemes.

Are default config values always static in this packages, or is it possible to make them code-generated per service?

Reply:
There will be more auth schemes implemented (e.g. `@httpBearerAuth`), so the ability to code generate different auth scheme lists based on the service model or customizations (like for SigV4A) will be needed.

How does the generic smithy-swift side generate default auth schemes currently?

Reply:
Changed as suggested. Now the default auth schemes are determined and code-generated into client config initializers during codegen.

Reply:
As for default auth schemes for smithy-swift side, the client config initializers for smithy clients are in fact missing completely at the moment. I've asked Andrew about it and that work is planned. Once he adds the codegen for client config for smithy-swift clients, after that (unless Andrew does as part of his PRs) I'll need to make additive changes.